From c3040b54d431e95c832505b9a20e859b465c3a51 Mon Sep 17 00:00:00 2001
From: Sai Vemuri <svemuri@redhat.com>
Date: Wed, 12 Nov 2014 19:08:13 -0500
Subject: [infinband] cxgb4: Add missing neigh_release in find_route

Message-id: <1415819300-32670-13-git-send-email-svemuri@redhat.com>
Patchwork-id: 99876
O-Subject: [RHEL7.1 PATCH BZ 1163467 12/19] RDMA/cxgb4: Add missing neigh_release in find_route
Bugzilla: 1163467
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>
RH-Acked-by: David S. Miller <davem@redhat.com>

Signed-off-by: Hariprasad Shenai <hariprasad@chelsio.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit d480201b2261d26b4ca1fa4cc2bfc6cb644733a3)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/cxgb4/cm.c b/drivers/infiniband/hw/cxgb4/cm.c
index 353e75d..bd8c0f1 100644
--- a/drivers/infiniband/hw/cxgb4/cm.c
+++ b/drivers/infiniband/hw/cxgb4/cm.c
@@ -417,6 +417,7 @@ static struct dst_entry *find_route(struct c4iw_dev *dev, __be32 local_ip,
   return NULL;
  if (!our_interface(dev, n->dev) &&
      !(n->dev->flags & IFF_LOOPBACK)) {
+  neigh_release(n);
   dst_release(&rt->dst);
   return NULL;
  }
-- 
1.7.1